src/components/Form/Form.tsx

```diff
@@ -4,7 +4,7 @@ import type { HTMLProps, ReactNode } from "react";
 
 import type { ClassName } from "types";
 
-export type BaseForm = {
```

Comment:
I've removed all the complicated logic for determining the conditional `Props` type because the generated `propTypes` were wrong. I think this is fine because we shouldn't really be having these weird union types anyway - they should be separate components. For now it'll be possible to use an "incompatible" combination of props but at least there won't be 70000 console errors.